idunn/blocks/opening_hour.py

```diff
@@ -1,13 +1,50 @@
-from apistar import validators
+import re
+import logging
+import datetime
+from datetime import datetime
+from pytz import country_timezones, timezone, UTC
 
+from apistar import validators
 from .base import BaseBlock
 
+import humanized_opening_hours as hoh
+from humanized_opening_hours.exceptions import ParseError, SpanOverMidnight
+from tzwhere import tzwhere
+
+"""
+We load the tz structure once when Idunn starts since it's a time consuming step
+"""
+tz = tzwhere.tzwhere(forceTZ=True)
+
+def get_tz(es_poi):
+    """
+    Returns the timezone corresponding to the coordinates of the POI.
+
+    >>> get_tz({}) is None
+    True
+
+    >>> get_tz({'coord':{"lon": None, "lat": 48.858260156496016}}) is None
+    True
+
+    >>> get_tz({'coord':{"lon": 2.2944990157640612, "lat": None}}) is None
+    True
+
+    >>> get_tz({'coord':{"lon": 2.2944990157640612, "lat": 48.858260156496016}})
+    <DstTzInfo 'Europe/Paris' LMT+0:09:00 STD>
+    """
+    if 'coord' in es_poi:
+        coord = es_poi.get('coord')
+        lon = coord.get('lon')
+        lat = coord.get('lat')
+        if lon is not None and lat is not None:
+            return timezone(tz.tzNameAt(lat, lon))
+    return None
 
 class OpeningHourBlock(BaseBlock):
     BLOCK_TYPE = 'opening_hours'
 
     status = validators.String()
```

Comment:
I would enumerate all possible status here (using `enum` param: https://docs.apistar.com/api-guide/type-system/#string)

And contrary to the original snippet, I think "open" and "closed" (instead of "close") would be more correct in proper english.

Reply:
done